x-pack/plugins/synthetics/public/apps/synthetics/state/root_reducer.ts

```diff
@@ -7,16 +7,30 @@
 
 import { combineReducers } from '@reduxjs/toolkit';
 
-import { syntheticsMonitorReducer } from './monitor_summary/synthetics_montior_reducer';
-import { monitorStatusReducer } from './monitor_summary';
-import { uiReducer } from './ui';
-import { indexStatusReducer } from './index_status';
-import { syntheticsEnablementReducer } from './synthetics_enablement';
-import { monitorListReducer } from './monitor_list';
-import { serviceLocationsReducer } from './service_locations';
-import { monitorOverviewReducer } from './overview';
+import {
+  syntheticsMonitorReducer,
+  SyntheticsMonitorState,
+} from './monitor_summary/synthetics_montior_reducer';
+import { monitorStatusReducer, MonitorSummaryState } from './monitor_summary';
+import { uiReducer, UiState } from './ui';
+import { indexStatusReducer, IndexStatusState } from './index_status';
+import { syntheticsEnablementReducer, SyntheticsEnablementState } from './synthetics_enablement';
+import { monitorListReducer, MonitorListState } from './monitor_list';
+import { serviceLocationsReducer, ServiceLocationsState } from './service_locations';
+import { monitorOverviewReducer, MonitorOverviewState } from './overview';
 
-export const rootReducer = combineReducers({
+export interface SyntheticsAppState {
+  ui: UiState;
+  indexStatus: IndexStatusState;
+  syntheticsEnablement: SyntheticsEnablementState;
+  monitorList: MonitorListState;
+  serviceLocations: ServiceLocationsState;
+  monitorStatus: MonitorSummaryState;
+  syntheticsMonitor: SyntheticsMonitorState;
+  overview: MonitorOverviewState;
+}
```

Comment:
@sphilipse I suppose this explicit typing was needed because redux toolkit's internal types aren't stable between versions?

Reply:
Yup, more recent versions led to a bunch of type errors. Explicit typing fixed it.

Reply:
Hi @awahab07, can you give this review a thumbs up or are there any changes you want to see?

Reply:
Hi, I'm a Redux Toolkit maintainer. I'm curious - what type errors did you see with this update? We actually specifically recommend inferring the root state from the result of `combineReducers` or `store.getState`, as you were doing previously. What broke here?